src/tokenizer.jl

```diff
@@ -61,7 +61,84 @@ function Tokenizer(tokenizer_path::String, vocab_size::Int)
     return Tokenizer(tokens, vocab_scores)
 end
 
-const BOS_TOKEN::Int32 = 1
+const BOS_TOKEN::Int32 = 2
+const EOS_TOKEN::Int32 = 3
+
+"""
+Encode the string text (input) into an upper-bound preallocated token array
+bos != 0 means prepend the BOS token (=1), eos != 0 means append the EOS token (=2)
```

Comment:
The second line does not make sense (bos, eos are not parameters).
Also, the array is not preallocated. Try to write your own docstring which describes what the function does.